spec/controllers/api/v1/products_controller_spec.rb

```diff
@@ -404,4 +404,44 @@ def make_request
       end
     end
   end
+
+  describe 'POST #transfer' do
+    def target_user
+      @target_user ||= create(:user)
+    end
+
+    def make_request(product, target_user)
```

Comment:
target_username maybe ?

Reply:
Remember that since on Rails there are no types (Duck typing) it's very important to give a good idea through names. target_user gives the idea of getting a User instance, which is correct in this case, however you are sending it on "email" key.